client/src/app/pages/applications/analysis-wizard/set-targets.tsx

```diff
@@ -36,12 +36,42 @@ export const SetTargets: React.FC = () => {
   const { watch, setValue } = useFormContext<AnalysisWizardFormValues>();
   const formTargets = watch("formTargets");
   const formRuleBundles = watch("formRuleBundles");
+  const formSources = watch("formSources");
+
+  const handleOnSelectedCardTargetChange = (
+    selectedRuleTarget: string,
+    selectedRuleBundle: RuleBundle
+  ) => {
+    const otherSelectedRuleTargets = formTargets.filter(
+      (formTarget) =>
+        !selectedRuleBundle.rulesets
+          .map((rule) => rule.metadata.target)
+          .includes(formTarget)
+    );
+    const definedSelectedTargets: string[] =
+      selectedRuleBundle.kind === "category"
+        ? [selectedRuleTarget]
+        : selectedRuleBundle.rulesets
+            .map((rulesets) => rulesets?.metadata?.target || "")
+            .filter((target) => !!target);
+
+    setValue("formTargets", [
+      ...otherSelectedRuleTargets,
+      ...definedSelectedTargets,
+    ]);
+  };
 
   const handleOnCardClick = (
     isSelecting: boolean,
     selectedRuleTarget: string,
     selectedRuleBundle: RuleBundle
   ) => {
+    const otherSelectedRuleSources = formSources.filter(
+      (formSource) =>
+        !selectedRuleBundle.rulesets
+          .map((rule) => rule.metadata.source)
+          .includes(formSource)
+    );
```

Comment:
We might not have much choices, but I wonder if we could avoid dealing with sources here.

Reply:
My concern is from a component viewpoint.
Set-targets need to know only about targets.
It would make sense to fetch ruleBundles in the wizard component and pass only the subset of targets to set-targets.tsx.